### ISSUE TITLE
admin pages abnormally slow to load

### ISSUE THREAD

**Issue by rmldj** [getgrav/grav-plugin-admin#1773]
I am running grav 1.6.16 and Admin v1.9.10 on linux Mint 18 (apache2 + php 7.3.9).
Switching between pages within the admin plugin takes abnormally long - even around 25s:

![gravadmin](https://user-images.githubusercontent.com/4105282/67640842-0ffa7f80-f8fd-11e9-96c1-d1a7a87c1f4f.png)

According to the debugger all this is spent during Render.

It seems that its getting worse (although I am not sure). I installed (and then completely removed) the mathjax plugin because it generated some error.

The normal site content loads on the other hand very fast - below 30ms. No problems here. So it must be something specific to the Admin plugin. I would be very grateful for help as currently its unusable.

**Comment by rhukster** [comment]
It’s kinda hard to work out what’s wrong with just the timeline. It just reinforces that render is taking very long.  Probably the simplest thing is to zip up your site and let me test it on my local.

**Comment by rmldj** [comment]
Hi,

Thanks for taking the time. I am attaching (through Google drive link) my
zipped site.
 gravsite.zip
<https://drive.google.com/file/d/1COYPs1wAISxRTrusRzretKc0dJYS8R_t/view?usp=drive_web>
After I logged out and again in (and being in different location) the times
have improved but are still abnormally slow for a local site (like 5s).
Again basically all the time is spent in the render stage.
Best,
Romuald

**Comment by rhukster** [comment]
Ok well it's something specific to your setup because on my local, your zip is fast:

<img width="1304" alt="monosnap-2019-10-28-11-45-50 image" src="https://user-images.githubusercontent.com/1084697/67703128-8d170a80-f978-11e9-8ec6-36fe524ed59c.png">

It really could be a wide variety of things, but it's probably something specific to your PHP setup. Some things to check:

* Give your PHP some more memory, 128M min
* If you are using APCu, give it some more shm_size memory.. I use 256M
* Make sure you have OPCache enabled/built in your PHP
* Do not use `open_basedir` this slows down PHP dramatically
* Disable Xdebug if you have that enabled, slows things down a lot too
* Disable any Profiling stuff (especially Xdebug profiler, it's really slow).

**Comment by rmldj** [comment]
That timeline on your local was a page within the admin plugin? My normal pages are very fast - I have problem with just the admin pages - so I cannot understand what could be so different.

It seems that I have all your suggestions implemented concerning PHP (I did not have APCu, which I installed but it did not help). I attach [phpinfo.pdf](https://github.com/getgrav/grav-plugin-admin/files/3780799/phpinfo.pdf)  with the ouput of my phpinfo().

Just for completeness - in the grav debugger messages for normal pages I have:
```
Grav v1.6.16
Environment Name: localhost
Cache: [true] Setting: [auto] Driver: [apcu]
Page cache hit.
Routed to page /notes/lattice-free-energy-from-ml (type: item)
```
while for the admin pages
```
Grav v1.6.16
Environment Name: localhost
Admin v1.9.10
Cache: [false] Setting: [auto] Driver: [apcu]
Page cache missed, rebuilding pages..
Routed to page (type: config)
```

**Comment by rhukster** [comment]
Yes, this is with your zip file, untouched.  I just removed your admin user and created my own.  I'm clicking the next/prev buttons in the admin.  Admin cache is off ( you can try enabling it in the admin plugin settings)

<img width="1295" alt="monosnap-2019-10-28-16-02-39 image" src="https://user-images.githubusercontent.com/1084697/67721550-6d91d900-f99c-11e9-8156-3814c3680ea8.png">

**Comment by rmldj** [comment]
I made a try of installing first version 1.8.20 then version 1.10 of the admin plugin. However my version of grav was perhaps incompatible and after logging in, I got an error that some template is missing. Hence I reverted back to the previous version (1.9.10) but got a similar error. So I **manually erased the subdirectories in the `cache/` folder**. This apparently solved the issue.

Now the admin pages are very fast (around 60-80ms). So perhaps there was some inconsistency/problem with the cached templates/files??

Although I guess it does not explain why you had good performance with the zipped version which included the `cache/` folder...

Anyway, if someone stumbles on this issue, I would suggest to try to manually erase the subdirectories in the `cache/` folder.
